### PR TITLE
Use Concurrent::FixedThreadPool to control compressors threads count

### PR DESCRIPTION
**Pull request by iggant**
When we have heavy compressors for gzip files assets precompile can end up with 500 or more threads, 
with this commit we restrict possible thread counts to processor count.

### CODE REVIEW

File: lib/sprockets/manifest.rb [hunk 1]
```diff
@@ -160,7 +160,7 @@ def compile(*args)
       end
 
       filenames              = []
-      concurrent_compressors = []
+      concurrent_compressors = Concurrent::FixedThreadPool.new( Concurrent.processor_count )
```
Comment (rafaelfranca) [inline lib/sprockets/manifest.rb]: Remove the space inside the parentheses

**Comment by rafaelfranca** [comment]
Makes sense. Thanks

**Comment by matthewd** [comment]
I daresay you have larger problems if you're outputting 500 compressible assets... but anyway, that's not how that API works.

**Comment by schneems** [comment]
Can you fix the style issue mentioned by @rafaelfranca. Also if you rebase against master your tests should pass. That should be good enough to get this across the line.

Thanks for the PR!

**Comment by schneems** [comment]
Can you fix the style issue mentioned by @rafaelfranca. Also if you rebase against master your tests should pass. That should be good enough to get this across the line.

**Comment by matthewd** [comment]
A rebase isn't going to fix this; it is Wrong.

**Comment by lucasallan** [comment]
@matthewd could you elaborate ?

**Comment by hansottowirtz** [comment]
I also tried this, it hangs. When I try to precompile assets (with Rails), an error shows up: `fatal: No live threads left. Deadlock?`. Very strange.

**Comment by hansottowirtz** [comment]
I modified the code a little bit, and it's working now. It is in #372 but don't mind copying it.

**Comment by lucasallan** [comment]
@hansottowirtz Could you try #376 and let me know if it works for you?

**Pull request merged**